### PR TITLE
docs: Initial API handler docs

### PR DESCRIPTION
**Pull request by c-ehrlich**
## 🎯 Changes

- Create initial API handler docs. Tried to keep it short and simple. I put it after the router and context docs as this makes the most sense to me.
- Link back to the API handler page from the context page

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [x] If necessary, I have added documentation related to the changes made.
- [x] I have added or updated the tests related to the changes made.

### CODE REVIEW

**Comment by c-ehrlich** [comment]
> > Opening as a draft because linking to a sub-section of a page seems not to scroll to that section when the page is clicked. Maybe my syntax is incorrect? Will try to get this fixed, but interested in feedback on the actual docs in the meantime.
> 
> Links seems to be working for me? You have an example of a broken one?
working for me as well now on the sample deployment, maybe issues related to my dev environment?

File: www/docs/server/api-handler.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+id: api-handler
+title: API Handler
+sidebar_label: API Handler
+slug: /api-handler
+---
+tRPC is not a backend of its own, but rather lives inside of other backends such as Next.js or Express. Despite that, most of tRPC's features and syntax are the same no matter which backend you are using. The API handler, also called Adapter in some cases, enables this by acting as the glue between HTTP requests to your backend and tRPC.
```
Comment (juliusmarminge) [inline www/docs/server/api-handler.md:7]: ... also called [adapter](https://trpc.io/docs/adapters)...

File: www/docs/server/api-handler.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+The API Handler receives a request from the server, uses the `createContext` function to generate [Context](./context), and then sends the request and context to the [Router](./router).
```
Comment (juliusmarminge) [inline www/docs/server/api-handler.md:9]: context, procedure(?)

**Comment by juliusmarminge** [comment]
Would also be nice to connect this https://trpc.io/docs/cors#enabling-on-the-server and explain that it can be wrappped etc

File: www/docs/server/context.md [hunk 3]
```diff
@@ -5,7 +5,7 @@ sidebar_label: Request Context
 slug: /context
 ---
 
-The `createContext()` function is called for each request, and the result is propagated to all resolvers. You can use this to pass contextual data down to the resolvers.
+The `createContext()` function is called for each request, and the result is propagated to all resolvers. You can use this to pass contextual data down to the resolvers.  This is done by including the context in the [API Handler](/docs/api-handler).
```
Comment (KATT) [inline www/docs/server/context.md:8]:
```suggestion
The `createContext()` function is called for each request, and the result is propagated to all resolvers. You can use this to pass contextual data down to the resolvers. This is done by including the context in the [API Handler](/docs/api-handler).
```

**Review by KATT — COMMENTED** [review]
Hey @c-ehrlich, could you have a look at julius' comments please? :)

**Comment by c-ehrlich** [comment]
> Would also be nice to connect this https://trpc.io/docs/cors#enabling-on-the-server and explain that it can be wrappped etc

Since Next.js is probably the most common way of using tRPC at this point, maybe it would be good to give an explicit example of this?

Something like: (but can maybe find a more "real" example)
```ts
const nextApiHandler = createNextApiHandler({
  router: appRouter,
  createContext,
});

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse
) {
  // do anything that modifies the request or response
  // example: set a cookie
  res.setHeader(
    "set-cookie",
    "yourParameter=someValue; path=/; samesite=lax; httponly;"
  );
  return nextApiHandler(req, res);
}
```

What do you think?

File: www/docs/server/api-handler.md (hunk 2 above)
Comment (c-ehrlich) [reply www/docs/server/api-handler.md]: https://github.com/trpc/trpc/pull/3341/commits/d23b09e2e310e4fc6e8f498f0e8fd65518b4cd44

File: www/docs/server/api-handler.md (hunk 1 above)
Comment (c-ehrlich) [reply www/docs/server/api-handler.md]: https://github.com/trpc/trpc/pull/3341/commits/ac670db9e6cfebf6d401291b1c414ab07afc9dc9

**Comment by juliusmarminge** [comment]
> (but can maybe find a more "real" example)

I think CORS is a valid example. I've used this in monorepos with Next.js as the backend and a 2nd Vite app for example:

CleanShot 2022-12-11 at 23 28 41@2x

**Comment by c-ehrlich** [comment]
Apologies for not touching this in two weeks, had a lot going on. Expanded on the API handler docs themselves, and also linked to them from the router and context pages as my experience from both tRPC and Theo discords is that many people struggle to understand the data flows.

File: www/docs/server/api-handler.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
+---
+id: api-handler
+title: API Handler
+sidebar_label: API Handler
+slug: /api-handler
+---
+
+tRPC is not a backend of its own, but rather lives inside of other backends such as Next.js or Express. Despite that, most of tRPC's features and syntax are the same no matter which backend you are using. The API handler, also called [adapter](/docs/adapters), enables this by acting as the glue between HTTP requests to your backend and tRPC.
+
+The API Handler sits on a route in your server (usually `/api/trpc`, but this is just a convention) and processes all requests to that route and its subroutes. It receives a request from the server, uses the `createContext` function to generate [context](./context), and then sends the request and context to a [procedure](./procedure) in the router.
```
Comment (juliusmarminge) [inline www/docs/server/api-handler.md:10]: Links here are broken according to build error on Vercel

Should be https://trpc.io/docs/procedures

**Review by juliusmarminge — CHANGES_REQUESTED** [review]

File: www/docs/server/api-handler.md (hunk 4 above)
Comment (juliusmarminge) [reply www/docs/server/api-handler.md:10]:
```suggestion
The API Handler sits on a route in your server (usually `/api/trpc`, but this is just a convention) and processes all requests to that route and its subroutes. It receives a request from the server, uses the `createContext` function to generate [context](./context), and then sends the request and context to a [procedure](./procedures) in the router.
```

**Review by juliusmarminge — APPROVED** [review]

File: www/docs/server/api-handler.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
+---
+id: api-handler
+title: API Handler
+sidebar_label: API Handler
+slug: /api-handler
+---
+
+tRPC is not a backend of its own, but rather lives inside of other backends such as Next.js or Express. Despite that, most of tRPC's features and syntax are the same no matter which backend you are using. The API handler, also called [adapter](/docs/adapters), enables this by acting as the glue between HTTP requests to your backend and tRPC.
```
Comment (sachinraja) [inline www/docs/server/api-handler.md:8]: 👏love this

File: www/docs/server/api-handler.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+  if (req.method === 'OPTIONS') {
+    res.writeHead(200);
+    return res.end();
+  }
+
+  // pass the (modified) req/res to the handler
+  return nextApiHandler(req, res);
+}
+```
```
Comment (sachinraja) [inline www/docs/server/api-handler.md:67]: This is great!!

**Review by sachinraja — APPROVED** [review]

**Pull request merged**